### PR TITLE
feat(encore): add `encore` spec

### PR DESCRIPTION
**Pull request by BogDAAAMN**
**What kind of change does this PR introduce?**
Adds completion specs for the [Encore](https://encore.dev/) CLI. See docs at https://encore.dev/docs/develop/cli-reference.

The spec is purely generated with Fig's cobra integration. See implementation here https://github.com/BogDAAAMN/encore/pull/1.

I am not yet too familiar with the Encore CLI yet, but I think it's a good start to have the commands auto-completed. Once I will use it more and find some good generators ideas I will implement them. I am also pinging @eandre and @bih, feel free to suggest generators (e.g. list `app-ids` or `service-names`), I would be happy implement them 😊

**Additional info**
Fig completing Encore CLI commands

### CODE REVIEW

File: src/encore.ts [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+            {
+              name: ["--lang", "-l"],
+              description:
+                'The language to generate code for ("typescript" and "go" are supported)',
+              args: { name: "lang" },
+            },
+            {
+              name: ["--output", "-o"],
+              description: "The filename to write the generated client code to",
+              args: { name: "output", template: ["filepaths"] },
```
Comment (fedeci) [inline src/encore.ts:182]:
```suggestion
              args: { 
                name: "output",
                template: ["filepaths"],
                suggestCurrentToken: true
              },
```

**Review by fedeci — APPROVED** [review]
Just one quick thing to fix then we can merge this! Thanks @BogDAAAMN!

File: src/encore.ts (hunk 1 above)
Comment (BogDAAAMN) [reply src/encore.ts:182]: Great suggestion! TIL about `suggestCurrentToken`, I thought it's always true 😳

**Comment by BogDAAAMN** [comment]
Thank you @fedeci, done 👍

File: src/encore.ts [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+                'The language to generate code for ("typescript" and "go" are supported)',
+              args: { name: "lang" },
+            },
+            {
+              name: ["--output", "-o"],
+              description: "The filename to write the generated client code to",
+              args: { 
+                name: "output",
+                template: ["filepaths"],
+                suggestCurrentToken: true
```
Comment (fedeci) [inline src/encore.ts:185]:
```suggestion
                suggestCurrentToken: true,
```

File: src/encore.ts [hunk 3]
```diff
@@ -0,0 +1,426 @@
...
+            {
+              name: ["--lang", "-l"],
+              description:
+                'The language to generate code for ("typescript" and "go" are supported)',
+              args: { name: "lang" },
+            },
+            {
+              name: ["--output", "-o"],
+              description: "The filename to write the generated client code to",
+              args: {
```
Comment (fedeci) [inline src/encore.ts:182]:
```suggestion
              args: {
```

**Pull request merged**